### ISSUE TITLE
Video training

### ISSUE THREAD

**Issue by Trapper4888** [QwenLM/Qwen2-VL#19]
In your dataset formatting for training example, you show an example with images.

Could you please show an example for video? Especially for the FPS feature.

**Comment by ShuaiBai623** [comment]
The format for video is consistent with that of images, except that <|video_pad|> is used. The recommended FPS (frames per second) is fps=2. However, llama-factory does not yet support video input.

**Comment by hiyouga** [comment]
We have plan to support the video training in few days, stay tuned 🚀

**Comment by mmmwhy** [comment]
> We have plan to support the video training in few days, stay tuned 🚀

@hiyouga wait for your jobs !

**Comment by mmmwhy** [comment]
@ShuaiBai623 I saw you close this issue, but there is no change about https://github.com/QwenLM/Qwen2-VL?tab=readme-ov-file#training part😂

Does there have any video training demo

**Comment by hiyouga** [comment]
@mmmwhy soon

**Comment by hiyouga** [comment]
Video training has been supported We have supported video finetuning https://github.com/hiyouga/LLaMA-Factory/tree/main/data#multimodal-video-dataset-1

**Comment by mmmwhy** [comment]
> Video training has been supported https://github.com/hiyouga/LLaMA-Factory/tree/main/data#multimodal-video-dataset-1

can i use video frames data to fine-tune, like this kind data
![image](https://github.com/user-attachments/assets/95d39d4c-1409-4b30-ad9f-0aab0329bb44)

**Comment by hiyouga** [comment]
@mmmwhy You only could use the dataset format described in llamafactory, we do not support the qwen2-vl's data format

**Comment by Trapper4888** [comment]
> @mmmwhy You only could use the dataset format described in llamafactory, we do not support the qwen2-vl's data format

So how is the fps and tokenization handled?

**Comment by hiyouga** [comment]
@Trapper4888 you need to specify `--video_fps 2.0`

**Comment by ApolloRay** [comment]
> @Trapper4888 you need to specify `--video_fps 1.0`

Hello, if I have a 750-frame video that needs training and inference, how should I set it? I set video_maxlen=500 and video_fps=16 in llamafactory, because if it is larger, it will cause OOM. I understand that the final frame extraction should take a comparison value of the video duration*fps and maxlen. In addition, how should I achieve this in the inference process of qwen2-vl?
